binary-oracle-pair/program/src/instruction.rs

```diff
@@ -120,6 +120,8 @@ pub fn deposit(
     program_id: &Pubkey,
     pool: &Pubkey,
     authority: &Pubkey,
+    user_transfer_authority: &Pubkey,
+    is_user_authority_signer: bool,
```

Comment:
Let's go with the concept that the user transfer authority will always be signing, so this param won't be needed

Reply:
If user will choose program account as an authority how can we sign transaction in such case ?
As I see we can’t create Keypair from the program account to produce signature

Reply:
I think no signature will be required in this case, right? @joncinque

Reply:
The whole idea is that using the same program account to control the program and also as a signer to transfer your funds is a potentially unsafe pattern, because anyone can get the program's signature if they call into the program.  This isn't particularly an issue for this program, but in general it's better to use ephemeral keypairs as a transfer authority